src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/EchoWebSocketHandler.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace NetCoreServer
+{
+    public class EchoWebSocketHandler
+    {
+        private const int MaxBufferSize = 128 * 1024;
+
+        public static async Task InvokeAsync(HttpContext context)
+        {
+            QueryString queryString = context.Request.QueryString;
+            bool  replyWithPartialMessages = queryString.HasValue && queryString.Value.Contains("replyWithPartialMessages");
+            bool replyWithEnhancedCloseMessage = queryString.HasValue && queryString.Value.Contains("replyWithEnhancedCloseMessage");
+
+            string subProtocol = context.Request.Query["subprotocol"];
+
+            if (context.Request.QueryString.HasValue && context.Request.QueryString.Value.Contains("delay10sec"))
+            {
+                Thread.Sleep(10000);
+            }
+
+            try
+            {
+                if (!context.WebSockets.IsWebSocketRequest)
+                {
+                    context.Response.StatusCode = 200;
+                    context.Response.ContentType = "text/plain";
+                    await context.Response.WriteAsync("Not a websocket request");
+
+                    return;
+                }
+
+                WebSocket socket;
+                if (!string.IsNullOrEmpty(subProtocol))
+                {
+                    socket = await context.WebSockets.AcceptWebSocketAsync(subProtocol);
+                }
+                else
+                {
+                    socket = await context.WebSockets.AcceptWebSocketAsync();
+                }
+
+                await ProcessWebSocketRequest(socket, replyWithPartialMessages, replyWithEnhancedCloseMessage);
+            }
+            catch (Exception)
+            {
+                // We might want to log these exceptions. But for now we ignore them.
```

Comment:
That may be hard for Azure service but perhaps simple print would be still beneficial. Unless there is easy way how to emit Kestrel error message.